File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        str_cln=re.sub(r'[^a-zA-Z0-9]','',s).lower()
        i,j=0,len(str_cln)-1
        while i<=j:
            if str_cln[i]!=str_cln[j]:
                return False
            i+=1
            j-=1 
        return True       

        

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^a-zA-Z0-9]","").toLowerCase();
        return palindrome(s,0,s.length()-1);
    }
    public boolean palindrome(String s,int i,int j){
        if(i>j) return true;
        if(s.charAt(i) != s.charAt(j)){
            return false;
        }
        return palindrome(s,i+1,j-1);
    }
}